ios-shoppe-demo/ios-shoppe-demo/ios-shoppe-demo/APIService.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  APIService.swift
+//  ios-shoppe-demo
+//
+//  Created on 3/4/20.
+//  Copyright © 2020 FullStory All rights reserved.
+//
+
+import Foundation
+
+class APIService {
+    static var sharedInstance = APIService()
+
+    var baseURL: String = "http://reactshoppe.s3-website-us-east-1.amazonaws.com/data/products.json"
+    var shopItems: [[String : Any]] = [] // TODO: Change this to be an array of class ShopItem
+
+    func getShoppeItem(completion: @escaping()->()) {
```

Comment:
You may not need the `@escaping` attribute since `completion()` is invoked within the function body. More info here: https://cocoacasts.com/what-do-escaping-and-noescape-mean-in-swift-3

Also: me new to Swift :)

Reply:
sweet im old school so i did not realize this changed 👍🏾

Reply:
This was in swift 3.0 but has since changed in 5.0 from what I can tell.

Reply:
Swift 3 made enclosures non-escaping by default and this remains true in swift 5: https://learnappmaking.com/escaping-closures-swift/

>In Swift 3 closures are made non-escaping by default. If you want to declare an escaping closure, for instance as a completion handler, you’ll have to mark it with @escaping.

It looks like, though, since I made my original comment the `completion` is now being used within a another closure: 
```URLSession.shared.dataTask(with: request) { (data, response, error) in ...```

From the article I linked to: "An escaping closure is a closure that’s called after the function it was passed to returns. In other words, it outlives the function it was passed to."

It's not clear to me if `completion()` is called after `getShoppeItem` returns. What are your thougts?

Reply:
I can read the article and focus particularly on the "The Risks Of Wrongly Escaping Closures". Is that what you recommending? Should i try to review this before the PR is approved?

Reply:
Yes, check the article out. The main concern avoiding memory leaks.  If you feel comfortable that your implementation is using `@escaping` correctly, then I'm good. Of course, feel free to reach out to Fred or another iOS engineer if it helps.

Reply:
Ah I understand that makes a lot of since I'll take it seriously and look make sure to educate my self on this. Ive done this a lot even at Apple and I know it works well the way I have done it, but It would be nice to completely understand at a deeper level. I'm going to try to learn this quick so it is no longer a blocker as well and I can move forward to try to reach our goal.

Reply:
Fortunately in this case I am using it correctly... however:

Based on the "The Risks Of Wrongly Escaping Closures" and "What’s Changed In Swift 3?".
- I now understand why swift closures are @non-escaping by default and the only reason the compiler is telling me I need it at the moment is because I'm calling URLSession's dataTask which is @escaping. (i did a test and removed it and it was fine without escaping if it was not there.)

- Going forward because non-escaping is more efficient I will leave it off unless I need to. For now it appears that it is required but it will be cool in the future to do non-escaping as to insure that as often as possible that the closure is called within the function its passed into before it actually returns.